### DISCUSSION TITLE
I18n shouldn't be non-case sensitive

### DISCUSSION THREAD

**Discussion by nicosommi** [vercel/next.js#40476] [Help]
Hi

Some [SEO experts](https://www.searchenginejournal.com/google-urls-are-case-sensitive/421264/) say the casing of the URL matters and having duplicates could be a serious issue for SEO (google may choose the wrong canonical for example). A duplicate means two pages with different cases (different HTTP resources as per the HTTP protocol definition) that return a 200 status code.

However, next.js is not sensitive to upper/lower case regarding the bundled i18n feature.
For example, both `es-ar` and `es-AR` return a 200 HTTP status code.
But also normalizes to the values provided on the values in the config, which means we literally can't understand the REAL URL used by the browser and then we can't even redirect in the edge (middleware).

Take a look at this screenshot that combines a debug value with a browser address bar:
![image](https://user-images.githubusercontent.com/1702755/189638406-983f0209-24e2-48b8-add8-5446ca4b4179.png)

As you can see, It doesn't send the actual URL written in the browser address bar, but the values from the config... this makes it impossible to implement a custom logic on the middleware without opting out of the i18n feature, literally enforcing a bad SEO practice.

Does this make sense to you? What's your opinion about it? Is it possible to provide the original URL somewhere? (We want to redirect to only one valid URL instead of replying with 200, lowercase in our case)

Thank you for reading :)

**Comment by dominiksipowicz** [comment]
It's probably better to give users control over which version to use (xx-xx vs. xx-XX) as per comment on linked issue [#31050](https://github.com/vercel/next.js/issues/31050).
> `pages` will be soon in bugfixes only, so we probably won't implement this (i18n is not supported in app-dir).
If this causes issues for you, you can redirect it yourself using middleware.
